dotnet/packages/Microsoft.Bot.Builder.M365/Microsoft.Bot.Builder.M365/AI/Prompt/SKFunctionWrapper.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.AI.TextCompletion;
+using Microsoft.SemanticKernel.Memory;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.Security;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.Bot.Builder.M365.AI.Prompt
+{
+    internal class SKFunctionWrapper<TState> : ISKFunction where TState : TurnState
+    {
+        // TODO: This is a hack around to get the default skill name from SK's internal implementation. We need to fix this.
+        public const string DefaultSkill = "_GLOBAL_FUNCTIONS_";
+
+        private string _name;
+        private PromptManager<TState> _promptManager;
+        private TState _turnState;
+        private ITurnContext _turnContext;
+        private IReadOnlySkillCollection? _skillCollection;
+
+        public string Name => _name;
+
+        public string SkillName => DefaultSkill;
+
+        public string Description => string.Empty;
+
+        public bool IsSemantic => false;
+
+        public bool IsSensitive => false;
+
+        public ITrustService TrustServiceInstance => throw new NotImplementedException();
+
+        public CompleteRequestSettings RequestSettings => throw new NotImplementedException();
+
+        public SKFunctionWrapper(ITurnContext turnContext, TState turnState, string name, PromptManager<TState> promptManager)
+        {
+            _name = name;
```

Comment:
same - can any of these be null?

Reply:
I'm adding a null check for arguments in the public methods, avoiding them for private and internal methods. 
There are a few assumptions in place:
1. Anything passed by the user will not be null since there will be a null check for that
2. If a dev (or something the dev did) did pass in a null value , there will be a "referencing null object" exception which should be caught by the tests.

Would love to discuss with you more on this.